### PR TITLE
V1.3: Add DG for Model package

### PR DESCRIPTION
**Pull request by SeeuSim**
- [x] Add basic documentation for the `Model` API, and its relevant classes.
- [x] Also, add the UML diagrams to better explain the code structure.

~TODO: Add UML diagrams~

Closes:  #120

### CODE REVIEW

**Comment by SeeuSim** [comment]
Please help review the class diagrams here: **[Fork Deployment](https://seeusim.github.io/tp/DeveloperGuide.html#model-component)**

Feel free to leave comments if necessary

File: docs/images/BetterModelClassDiagram.png
Comment (alson001) [inline docs/images/BetterModelClassDiagram.png:1]: show * multiplicity from RecipeBook to UniqueRecipeList for consistency?

**Review by alson001 — APPROVED** [review]
LGTM

File: docs/images/ModelClassDiagram.png
Comment (ZHTang29) [inline docs/images/ModelClassDiagram.png:1]: Looks good and is an accurate representation of the model package! Especially like the multiplicity for all the `RecipePortion`, `RecipeDuration` etc.

**Review by ZHTang29 — APPROVED** [review]
LGTM good work

File: docs/images/ModelClassDiagram.png
Comment (JamesLiuZX) [inline docs/images/ModelClassDiagram.png:1]: Looks great to me. No issues with the multiplicity and the dependencies for Ingredient

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -5,17 +5,20 @@ skinparam arrowColor MODEL_COLOR
...
+Recipe -up-> "*" Ingredient
 
+Recipe *--> RecipePortion
+Recipe *--> RecipeDuration
 Recipe *--> Name
-Recipe *--> Phone
-Recipe *--> Email
-Recipe *--> Address
+Recipe *--> Step
+Recipe *--> "*" IngredientQuantifier
```
Comment (JamesLiuZX) [inline docs/diagrams/BetterModelClassDiagram.puml:22]: Cherry-picking but ordering could either follow order of user input for intuitiveness or alphabetical

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -5,6 +5,51 @@ title: Developer Guide
...
+  separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Recipe>` instance that 
+  can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the 
+  list changes.
+* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a 
+  `ReadOnlyUserPref` object.
+* does not depend on any of the other three components (as the `Model` represents data entities of the domain, they 
+  should make sense on their own without depending on other components)
+* However, it is worth noting that clients need only pass valid `IngredientBuilder` instances to `Recipe` objects, 
+  from which the population of the `Ingredient` and `IngredientQuantifier` fields will be done by the `Recipe` class
+  and its methods.
```
Comment (JamesLiuZX) [inline docs/DeveloperGuide.md:37]: The sentence from `from which` can be a bit harder to read. Can try ", where the `Recipe` class and its methods then populate the `Ingredient` and `IngredientQuantifier` fields".

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 1 above)
Comment (SeeuSim) [reply docs/diagrams/BetterModelClassDiagram.puml:22]: Limitation of pUML, can't be fixed

**Pull request merged**